Add "analyze-instrumentation" command to dd-trace diagnostic tool (#3408)
DataDog/dd-trace-dotnet: docs/InstrumentationVerification/InstrumentationVerification.md

1. Verify that the environment variable `DD_TRACE_WRITE_INSTRUMENTATION_TO_DISK` is set to “true” (it should be on by default).
2. Run your application. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}
3. In order to run the analysis:
    1. To analyze a running process:

>         Via command line, run the command: `dd-trace analyze-instrumentation [process name] [pid]`

Reviewer: Shouldn't this one be just the PID? I thought specifying both is just for historic runs?

Author: Fixed in 59dc3c9f22d7700ec54e60483dd11a150535c39c

    2. To analyze a process that is no longer running (crashed/terminated):

>         Via command line, run the command: `dd-trace analyze-instrumentation [process name]`

Author: Fixed in 59dc3c9f22d7700ec54e60483dd11a150535c39c

>         **Note:** In this case the analysis will perform on the most recent log directory. To analyze a specific process, provide also the process ID.

Reviewer: ```suggestion
```

This kind of goes without saying I think?

Author: Fixed in 59dc3c9f22d7700ec54e60483dd11a150535c39c

> 2. Run or restart the application we are instrumenting. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}

Reviewer: We can not delete step 1, because it is on by default, right?

Author: Fix in d6e34e581c70cc2271806c12db2c3d4f2caf78f1

> 2. Run or restart the application we are instrumenting. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}

Reviewer: ```suggestion
2. Run your application. Observe that as we instrument the app, more and more files will appear under [Datadog Logs Folder]/InstrumentationVerification/{ProcessName}_{ProcessID}_{ProcessCreationTime}
```

Author: Fix in d6e34e581c70cc2271806c12db2c3d4f2caf78f1

3. Via command line, run the command: 

>     Datadog.InstrumentedAssemblyVerification.Standalone.exe [Full path of the folder generated in step 2] 

Reviewer: These are two distinct options, right? So maybe break them apart?

Also, if I understand correctly, the option of using the full path to the folder that was generated in step (2), which you deleted, is the only way we support running the analysis on a process that already crashed, right? So perhaps we need to bring it back and explain that distinction?

Author: Fix in d6e34e581c70cc2271806c12db2c3d4f2caf78f1
